### ISSUE TITLE
Publish Occam camera feeds through a ROS topic

### ISSUE THREAD

**Issue by ikemole** [correll/advancedrobotics#62]
I just completed the process of publishing images from the Occam Camera to a ROS topic. To do this, I adapted from one of the provided examples that displays the image to the screen to instead publish a topic. This way, any node can subscribe to it to either display or run SLAM. 

The code is here: `~/image_transport_ws/src/occam_publisher`
To launch the node:  `rosrun occam_publisher occam_publisher`
This publishes to the topic: `/occam/image`

More info can be found on the [wiki](https://github.com/correll/advancedrobotics/wiki#using-occam-images-in-ros).

**Comment by ikemole** [comment]
There is one problem. The `occam_publisher` node only works correctly when I run it directly from the Tegra by attaching a screen and keyboard to it. If I log in through SSH and run it, it doesn't work instead it just exits without even throwing an error.   

This is surprising because the ROS node does not actually need to display the image that it is publishing yet it only works when there is a screen attached. Any suggestion on this?

**Comment by ikemole** [comment]
I have rectified the issue. The problem was that the Occam driver has an option to use OpenGL enhancement and so it always tried to initialize OpenGL before reading images from the camera. When a screen is attached, this initialization works, but it fails when a screen is not detected and so the program exits. The option to use OpenGL is specified in the CmakeLists file (`~/image_transport_ws/src/occam_publisher/CMakeLists.txt`). I commented out the line that sets the `Use_OpenGL` option and everything works fine. I can now launch the occam_publisher through SSH.

I can subscribe to it from my computer and display it: 
![screenshot from 2015-11-03 18 23 04](https://cloud.githubusercontent.com/assets/2020624/10926851/27d40290-8258-11e5-8ff5-996a37a8b304.png)

It can also be seen on a web browser using a ROS package `web_video_server`:
![screenshot from 2015-11-03 18 27 01](https://cloud.githubusercontent.com/assets/2020624/10926916/9e8360ac-8258-11e5-951b-92cb6c529303.png)
